discojs/src/dataset/dataset.ts

```diff
@@ -1,6 +1,177 @@
-import type tf from '@tensorflow/tfjs'
+import { List } from "immutable";
 
-/**
- * Convenient type for the common dataset type used in TF.js.
- */
-export type Dataset = tf.data.Dataset<tf.TensorContainer>
+type DatasetLike<T> =
+  | AsyncIterable<T>
+  | Iterable<T>
+  // generators
+  | (() => AsyncIterator<T, void>)
+  | (() => Iterator<T, void>);
+
+/** Immutable serie of data */
```

Comment:
```suggestion
/** Immutable series of data */
```
Series always has an s even when there's a single series...

Reply:
human languages will never stop astonishing me :eye_speech_bubble: